src/wp-includes/blocks/template-part.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Server-side rendering of the `core/template-part` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/template-part` block on the server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string The render.
+ */
+function render_block_core_template_part( $attributes ) {
+	static $seen_ids = array();
+
+	$template_part_id = null;
+	$content          = null;
+	$area             = WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
+
+	if (
+		isset( $attributes['slug'] ) &&
+		isset( $attributes['theme'] ) &&
+		wp_get_theme()->get_stylesheet() === $attributes['theme']
+	) {
+		$template_part_id    = $attributes['theme'] . '//' . $attributes['slug'];
+		$template_part_query = new WP_Query(
+			array(
+				'post_type'      => 'wp_template_part',
+				'post_status'    => 'publish',
+				'post_name__in'  => array( $attributes['slug'] ),
+				'tax_query'      => array(
+					array(
+						'taxonomy' => 'wp_theme',
+						'field'    => 'slug',
+						'terms'    => $attributes['theme'],
+					),
+				),
+				'posts_per_page' => 1,
+				'no_found_rows'  => true,
+			)
+		);
+		$template_part_post  = $template_part_query->have_posts() ? $template_part_query->next_post() : null;
+		if ( $template_part_post ) {
+			// A published post might already exist if this template part was customized elsewhere
+			// or if it's part of a customized template.
+			$content    = $template_part_post->post_content;
+			$area_terms = get_the_terms( $template_part_post, 'wp_template_part_area' );
+			if ( ! is_wp_error( $area_terms ) && false !== $area_terms ) {
+				$area = $area_terms[0]->name;
+			}
+		} else {
+			// Else, if the template part was provided by the active theme,
+			// render the corresponding file content.
+			$template_part_file_path = get_stylesheet_directory() . '/block-template-parts/' . $attributes['slug'] . '.html';
+			if ( 0 === validate_file( $attributes['slug'] ) && file_exists( $template_part_file_path ) ) {
+				$content = _inject_theme_attribute_in_block_template_content( file_get_contents( $template_part_file_path ) );
```

Comment:
`file_get_contents()` returns `false` on failure. `_inject_theme_attribute_in_block_template_content()` expects a `string` data type. Within the deeper logic (i.e. in `WP_Block_Parser::next_token()) the content is passed as the 2nd argument to `preg_match`. That argument requires a non-nullable string.

To avoid possible PHP 8.1 deprecation notices and future PHP 9 fatal error, I'd suggest:

```suggestion
				$content = file_get_contents( $template_part_file_path );
				$content = is_string( $content ) && '' !== $content
					? _inject_theme_attribute_in_block_template_content(  $content )
					: '';
```

Also a micro-optimization to avoid further processing of an empty value.

Reply:
Note: May want to change the empty string value to `null` as the logic just below this code block bails out if there's no content.

Reply:
This will need to be changed in Gutenberg as this file is copied from `@wordpress/block-library`.

Reply:
> This will need to be changed in Gutenberg as this file is copied from `@wordpress/block-library`.

Yeah, this is sync'd from the `@wordpress/block-library` package upon build. For this reason, I won't apply the fix here but file a GB PR.

Reply:
https://github.com/WordPress/gutenberg/pull/36324